### PR TITLE
PPTP-1141 GRS Callback 

### PR DESCRIPTION
**Pull request by kiranprakashdigital**
### Description of Work carried through

Brief description of what/why/how.

#### Check list 
 - [x] `./precheck` was executed (Integration/Component/Unit tests)
 - [x] `sbt scalafmt test:scalafmt` was executed

### CODE REVIEW

**Review by SteveSugden — APPROVED** [review]
LGTM

File: test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/group/GroupMemberGrsControllerSpec.scala [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+    "store the returned business entity information" when {
+      "registering UK Limited Company group member" in {
+        await(simulateLimitedCompanyCallback())
+
+        val memberDetails = getLastSavedRegistration.groupDetail.get.members.last
+
+        memberDetails.organisationDetails mustBe groupMemberDetailsUKCompany.organisationDetails
+        memberDetails.addressDetails mustBe groupMemberDetailsUKCompany.addressDetails
+      }
```
Comment (RyanKellyHMRC) [inline test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/group/GroupMemberGrsControllerSpec.scala:79]: This test is pretty difficult to reason about. It relies on the fact that test data defined elsewhere is consistent; essentially that `PptTestData.groupMemberDetailsUKCompany` and `PptTestData.groupMemberDetailsUKCompany` are built with the same pieces of test data. 

I think we could improve tests of this nature by making it clearer to the reader that what's being checked is the same thing. In this case, perhaps, by constructing the group member details in the test from constituent parts of the `incorporationDetails` which is used by the involved mocks.

@kiranprakashdigital please don't see this as a criticism. This is a pet hate of mine and I know there are other examples elsewhere and, very probably, bad examples written by me.

**Review by RyanKellyHMRC — CHANGES_REQUESTED** [review]
A few minor suggestions to improve readability.

File: app/uk/gov/hmrc/plasticpackagingtax/registration/models/genericregistration/IncorporationAddressDetails.scala [hunk 2]
```diff
@@ -63,6 +64,27 @@ case class IncorporationAddressDetails(
     }
   }
 
+  def toGroupAddressDetails: AddressDetails = {
+    val premises: Option[String] = if (this.premises.getOrElse("").nonEmpty) this.premises else None
+    premises match {
+      case Some(value) =>
```
[the remark this answers is not in the document]
Comment (SteveSugden) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/models/genericregistration/IncorporationAddressDetails.scala:70]: If the intention is to trim the optional value then you could just premises.map(_.trim) (I think)

[the remark this answers is not in the document]
Comment (SteveSugden) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/models/genericregistration/IncorporationAddressDetails.scala:70]: Wrong thread!

File: app/uk/gov/hmrc/plasticpackagingtax/registration/models/genericregistration/IncorporationAddressDetails.scala [hunk 3]
```diff
@@ -63,6 +64,27 @@ case class IncorporationAddressDetails(
     }
   }
 
+  def toGroupAddressDetails: AddressDetails = {
+    val premises: Option[String] = if (this.premises.getOrElse("").nonEmpty) this.premises else None
```
[the remark this answers is not in the document]
Comment (SteveSugden) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/models/genericregistration/IncorporationAddressDetails.scala:68]: If the intention is to trim the optional value then you could just premises.map(_.trim) (I think)

[the remark this answers is not in the document]
Comment (kiranprakashdigital) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/models/genericregistration/IncorporationAddressDetails.scala]: Fixed it.

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/group/GroupMemberGrsController.scala [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    getDetails: String => Future[IncorporationDetails]
+  )(implicit
+    hc: HeaderCarrier,
+    request: JourneyRequest[AnyContent]
+  ): Future[Either[ServiceError, Registration]] =
+    for {
+      details <- getDetails(journeyId)
+      result <- update { registration =>
+        val updatedGroupDetails: GroupDetail = registration.groupDetail match {
+          case Some(value) =>
```
[the remark this answers is not in the document]
Comment (kiranprakashdigital) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/group/GroupMemberGrsController.scala]: Done

File: app/uk/gov/hmrc/plasticpackagingtax/registration/models/genericregistration/IncorporationAddressDetails.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (kiranprakashdigital) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/models/genericregistration/IncorporationAddressDetails.scala]: Done

File: test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/group/GroupMemberGrsControllerSpec.scala (hunk 1 above)
Comment (kiranprakashdigital) [reply test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/group/GroupMemberGrsControllerSpec.scala:80]: I have changed the test to compare to `incorporaationDetails` companyName and addressDetails.

**Review by RyanKellyHMRC — APPROVED** [review]
Thanks!

**Pull request merged**